data/stages.lua

```diff
@@ -0,0 +1,33 @@
+-- set to true to enable experience stages
+stagesEnabled = false
+
+experienceStages = {
+	{minlevel = 1, maxlevel = 8, multiplier = 7},
+	{minlevel = 9, maxlevel = 20, multiplier = 6},
+	{minlevel = 21, maxlevel = 50, multiplier = 5},
+	{minlevel = 51, maxlevel = 100, multiplier = 4},
+	{minlevel = 101, multiplier = 3}
+}
+
+stagesMap = {}
+useLastStageLevel = false
+lastStageLevel = 0
+
+for _, stage in ipairs(experienceStages) do
```

Comment:
I actually dont understand all of this code, why not just search thru table directly?

Reply:
so that no iteration is necessary every time a player gains experience. So with this cache, we can get the multiplier using the player level as index. this Lua code does exactly the same cache as the old C ++ code does.

```lua
-- stagesMap dump
{
    [1] = 7,
    [2] = 7,
    [3] = 7,
    [4] = 7,
    [5] = 7,
    [6] = 7,
    [7] = 7,
    [8] = 7,
    [9] = 6,
    [10] = 6,
    [11] = 6,
    [12] = 6,
    [13] = 6,
    [14] = 6,
    [15] = 6,
    [16] = 6,
    [17] = 6,
    [18] = 6,
    [19] = 6,
    [20] = 6,
    [21] = 5,
    [22] = 5,
    [23] = 5,
    [24] = 5,
    [25] = 5,
    [26] = 5,
    [27] = 5,
    [28] = 5,
    [29] = 5,
    [30] = 5,
    [31] = 5,
    [32] = 5,
    [33] = 5,
    [34] = 5,
    [35] = 5,
    [36] = 5,
    [37] = 5,
    [38] = 5,
    [39] = 5,
    [40] = 5,
    [41] = 5,
    [42] = 5,
    [43] = 5,
    [44] = 5,
    [45] = 5,
    [46] = 5,
    [47] = 5,
    [48] = 5,
    [49] = 5,
    [50] = 5,
    [51] = 4,
    [52] = 4,
    [53] = 4,
    [54] = 4,
    [55] = 4,
    [56] = 4,
    [57] = 4,
    [58] = 4,
    [59] = 4,
    [60] = 4,
    [61] = 4,
    [62] = 4,
    [63] = 4,
    [64] = 4,
    [65] = 4,
    [66] = 4,
    [67] = 4,
    [68] = 4,
    [69] = 4,
    [70] = 4,
    [71] = 4,
    [72] = 4,
    [73] = 4,
    [74] = 4,
    [75] = 4,
    [76] = 4,
    [77] = 4,
    [78] = 4,
    [79] = 4,
    [80] = 4,
    [81] = 4,
    [82] = 4,
    [83] = 4,
    [84] = 4,
    [85] = 4,
    [86] = 4,
    [87] = 4,
    [88] = 4,
    [89] = 4,
    [90] = 4,
    [91] = 4,
    [92] = 4,
    [93] = 4,
    [94] = 4,
    [95] = 4,
    [96] = 4,
    [97] = 4,
    [98] = 4,
    [99] = 4,
    [100] = 4,
    [101] = 3
}
```